server/src/main/java/org/elasticsearch/cluster/AckedClusterStateUpdateTask.java

```diff
@@ -29,18 +30,20 @@
  * An extension interface to {@link ClusterStateUpdateTask} that allows to be notified when
  * all the nodes have acknowledged a cluster state update request
  */
-public abstract class AckedClusterStateUpdateTask<Response> extends ClusterStateUpdateTask implements AckedClusterStateTaskListener {
+public abstract class AckedClusterStateUpdateTask extends ClusterStateUpdateTask implements AckedClusterStateTaskListener {
 
-    private final ActionListener<Response> listener;
+    private final ActionListener<AcknowledgedResponse> listener;
```

Comment:
We could use `? extends` here as well so a cast is not necessary?

Reply:
That won't work unfortunately. If we bound by `? extends` then it could be that the listener is actually a listener for some subclass of `AcknowledgedResponse` so that gives a compile time error ... it always seems like this should work though :)